File: finafanBE/src/main/java/com/a702/finafanbe/core/ble/dto/response/MatchFansResponse.java
package com.a702.finafanbe.core.ble.dto.response;

import com.a702.finafanbe.core.user.entity.User;

public record MatchFansResponse(
        Long userId,
        String userName,
        String imgUrl
) {
    public static MatchFansResponse of(User user, String imgUrl){
        return new MatchFansResponse(
                user.getUserId(),
                user.getName(),
                imgUrl
        );
    }
}

File: finafanAndroid/app/src/main/java/com/a702/finafan/data/ble/dto/response/MatchFansResponse.kt
package com.a702.finafan.data.ble.dto.response

import com.a702.finafan.domain.ble.model.Fan

data class MatchFansResponse(
    val userId: Long,
    val name: String,
    val profileImageUrl: String
)

fun MatchFansResponse.toDomain(): Fan = Fan(
    id = userId,
    name = name,
    profileUrl = profileImageUrl
)